### ISSUE TITLE
resource: file support for SAML is broken

### ISSUE THREAD

**Issue by leleuj** [apereo/cas#2010]
The path to the file is not properly computed when using the `resource:` prefix.

```
Caused by: org.pac4j.saml.exceptions.SAMLException: Could not create keystore
	at org.pac4j.saml.client.SAML2ClientConfiguration.createKeystore(SAML2ClientConfiguration.java:472)
	at org.pac4j.saml.client.SAML2ClientConfiguration.<init>(SAML2ClientConfiguration.java:131)
	at org.pac4j.saml.client.SAML2ClientConfiguration.<init>(SAML2ClientConfiguration.java:95)
	at org.pac4j.config.client.PropertiesConfigFactory.tryCreateSaml2Client(PropertiesConfigFactory.java:223)
	at org.pac4j.config.client.PropertiesConfigFactory.build(PropertiesConfigFactory.java:115)
	at org.apereo.cas.support.pac4j.config.Pac4jConfiguration.builtClients(Pac4jConfiguration.java:218)
....
Caused by: java.io.FileNotFoundException: /Users/bidou/sources/cas-pac4j-oauth-demo/ile:/Users/bidou/sources/cas-pac4j-oauth-demo/cas-pac4j-oauth-client-demo/target/cas.war!/WEB-INF/classes!/samlKeystore.jks (No such file or directory)
	at java.io.FileOutputStream.open0(Native Method)
	at java.io.FileOutputStream.open(FileOutputStream.java:270)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
```

**Comment by leleuj** [comment]
@mmoayyed this is a show stopper for RC3 release: I will work as soon as possible.

**Comment by leleuj** [comment]
So I have a fix for this issue: https://github.com/pac4j/pac4j/pull/688

**This problem proves you right on the fact we should use Spring**: using the `File` object for resources inside JARs does not work.

The best I can do is disable the Keystore creation to avoid blocking the CAS startup. Then, the SAML authentication delegation will only work if you define a file path (no `resource:` or `classpath:`). For me, it's fairly acceptable until we work on that for pac4j v2.0.0.

I just added this future change in the pac4j roadmap I'll share soon.

If we agree, I'll release the pac4j 1.9.3 version and  the spring-webmvc-pac4j 1.1.3 version to upgrade CAS 5.0.0.RC3.

**Comment by mmoayyed** [comment]
How would you "disable the Keystore creation" ? We might document this shortcoming that if one wishes to let pac4j create the keystore, then there may be a few problems until such as such gets fixed. Is there more to it than that?

**Comment by leleuj** [comment]
I mean that the keystore creation is now defined by a `boolean` property. Though, I'm not sure this is the right approach because even if it doesn't block the CAS server startup, the SAML authentication delegation is still broken.

I guess I should keep the keystore creation as it is and add some warning on the CAS documentation to say that the keystore path must be specified as a relative or absolute file, not as a resource file.

Until we fix that in 2.0.0 (or why not in 1.9.4).

**Comment by mmoayyed** [comment]
> add some warning on the CAS documentation to say that the keystore path must be specified as a relative or absolute file, not as a resource file.

Agreed. I suspect most would want to define it as an absolute path anyway.

**Comment by leleuj** [comment]
My conclusion is there is no emergency to fix that. I'll add a better error message in pac4j 1.9.3.

I'll submit a PR to add a warning on CAS documentation as well.

Maybe we'll remove the resource: or prefix: prefixes for the 2.0.0 release.

**Comment by leleuj** [comment]
PR: https://github.com/apereo/cas/pull/2016

**Comment by leleuj** [comment]
I have another issue:

```
java.lang.NullPointerException
	at org.pac4j.saml.client.SAML2ClientConfiguration.getServiceProviderMetadataPath(SAML2ClientConfiguration.java:339) ~[pac4j-saml-1.9.2.jar!/:?]
	at org.pac4j.saml.metadata.SAML2ServiceProviderMetadataResolver.<init>(SAML2ServiceProviderMetadataResolver.java:59) ~[pac4j-saml-1.9.2.jar!/:?]
	at org.pac4j.saml.client.SAML2Client.initServiceProviderMetadataResolver(SAML2Client.java:155) ~[pac4j-saml-1.9.2.jar!/:?]
```

If the service provider metadata path is not set (like `cfg.setServiceProviderMetadataPath`), it generates a `NullPointerException`.

I will provide a simple fix for pac4j v1.9.3: https://github.com/pac4j/pac4j/issues/690 and retest the SAML delegation authentication in CAS.
